sdk/python/generator/Generator.py

```diff
@@ -10,122 +10,89 @@
 from .PodTypeFormatter import PodTypeFormatter
 from .printers import BuiltinPrinter, create_pod_printer
 from .StructTypeFormatter import StructFormatter
-from .type_objects import BaseObject, StructObject
 from .TypeFormatter import TypeFormatter
 
 
-class Field:
-	def __init__(self, ast_model, type_instance, printer):
-		self.ast_model = ast_model
-		self.type_ast_model = type_instance.ast_model
-
+class AstExtensions:
+	def __init__(self, type_model, printer):
+		self.type_model = type_model
 		self.printer = printer
+
+		self.is_contents_abstract = False
 		self.bound_field = None
 		self.size_fields = []
 
-	def get_printer(self):
-		return self.printer
-
-	def set_bound_field(self, other_field):
-		self.bound_field = other_field
-
-	def is_bound(self):
-		return self.bound_field is not None
 
-	def add_sizeof_field(self, other_field):
-		self.size_fields.append(other_field)
-
-
-def to_type_formatter_instance(type_instance):
+def to_type_formatter_instance(ast_model):
 	type_formatter_class = {
 		DisplayType.STRUCT: StructFormatter,
 		DisplayType.ENUM: EnumTypeFormatter,
 		DisplayType.BYTE_ARRAY: PodTypeFormatter,
 		DisplayType.INTEGER: PodTypeFormatter
-	}[type_instance.ast_model.display_type]
-
-	return type_formatter_class(type_instance)
-
-
-def to_virtual_type_instance(ast_model):
-	object_class = {
-		DisplayType.STRUCT: StructObject,
-		DisplayType.ENUM: BaseObject,
-		DisplayType.BYTE_ARRAY: BaseObject,
-		DisplayType.INTEGER: BaseObject
 	}[ast_model.display_type]
 
-	return object_class(ast_model)
+	return type_formatter_class(ast_model)
 
 
-def process_concrete_struct(type_map, struct_type, abstract_impl_map):
-	factory_type = type_map[struct_type.ast_model.factory_type]
+def process_concrete_struct(type_map, struct_model, abstract_impl_map):
+	factory_type = type_map[struct_model.factory_type]
 
 	discriminators = [] if not hasattr(factory_type, 'discriminator') else factory_type.discriminator
 	for i, discriminator in enumerate(discriminators):
 		discriminators[i] = discriminator
 
-	abstract_impl_map.add(factory_type, struct_type)
+	abstract_impl_map.add(factory_type, struct_model)
 
 
-def bind_size_fields(struct_type):
+def bind_size_fields(struct_model):
 	# go through structs and bind size fields to arrays
-	for field in struct_type.layout:
-		if field.ast_model.display_type.is_array and isinstance(field.ast_model.size, str):
-			size_field_name = field.ast_model.size
-			size_field = struct_type.get_field_by_name(size_field_name)
-			size_field.set_bound_field(field)
-
-		if field.ast_model.is_size_reference:
-			struct_field = struct_type.get_field_by_name(field.ast_model.value)
-			field.set_bound_field(struct_field)
+	for field_model in struct_model.fields:
+		if field_model.display_type.is_array and isinstance(field_model.size, str):
+			size_field_name = field_model.size
+			size_field_model = next(field2_model for field2_model in struct_model.fields if size_field_name == field2_model.name)
```

Comment:
would extract the inline (same below)

(or maybe better: move it to AstExtensions:?)